mesonbuild/dependencies/dev.py

```diff
@@ -129,21 +129,19 @@ class LLVMDependency(ExternalDependency):
         'llvm-config-3.5', 'llvm-config35',
         'llvm-config-5.0', 'llvm-config-devel', # development snapshot
     ]
-    llvmconfig = None
-    _llvmconfig_found = False
-    __best_found = None
```

Comment:
IIRC these were stored on the class instance to avoid having to run `check_llvmconfig` multiple times in the same configure session for different `dependency('llvm', ...)` calls. Since `LLVMDependency` can be called multiple times with different `modules:` kwargs, I think it might be worth keeping it around.

Reply:
But that prevents us from checking for multiple versions of LLVM (a FIXME that you left). I was thinking that we could cut the middle by caching the actuall llvm-configs that we've found so we could avoid multiple subprocess calls, but still get different versions if we need them.

Reply:
Good point. I hadn't considered that the same system might have multiple `llvm-config*` executables that might be needed to match multiple version conditions. The subprocess calls can indeed be cached out later (I'll work on that when I have some time), so it will be fine overall.

Reply:
Just to be clear, by "multiple version conditions", I meant things like: `version: ['>=2.0', '<3.0']`, that doesn't seem to be what you implemented here?

Reply:
But again, I don't know if anyone will ever want to do that with llvm, so maybe that can wait till someone complains. ;)

Reply:
I have a working patch to cache the llvm-config version to file. I have mixed feelings about it, since it'll be a win if you need to call `dependency(llvm)` multiple times, but it introduces some cost otherwise.

Reply:
I have a patch that stores the outputs of `Popen_safe` calls in the coredata which gets cached between runs. That will supersede that, so I think this is ok as-is.

Reply:
That sounds even better than the class method caching that the LLVM dependency does now.